packages/fxa-settings/src/components/LinkExpired/index.tsx

```diff
@@ -0,0 +1,84 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+import React from 'react';
+import { FtlMsg } from 'fxa-react/lib/utils';
+
+type LinkExpiredProps = {
+  linkType: LinkType;
+};
+
+type LinkType = 'reset-password' | 'signin';
+
+const resendResetPasswordLinkHandler = () => {
+  //   TODO: add resend link action
+};
+
+const resendSigninLinkHandler = () => {
+  //   TODO: add resend link action
+};
+
+const getTemplateValues = (linkType: LinkType) => {
+  let templateValues = {
+    headerText: '',
+    headerId: '',
+    messageText: '',
+    messageId: '',
+    resendLinkHandler: () => {},
+  };
+  switch (linkType) {
+    case 'reset-password':
+      templateValues.headerText = 'Reset password link expired';
+      templateValues.headerId = 'reset-pwd-link-expired-header';
+      templateValues.messageText =
+        'The link you clicked to reset your password is expired.';
+      templateValues.messageId = 'reset-pwd-link-expired-message';
+      templateValues.resendLinkHandler = resendResetPasswordLinkHandler;
+
+      break;
+    case 'signin':
+      templateValues.headerText = 'Confirmation link expired';
+      templateValues.headerId = 'signin-link-expired-header';
+      templateValues.messageText =
+        'The link you clicked to confirm your email is expired.';
+      templateValues.messageId = 'signin-link-expired-message';
+      templateValues.resendLinkHandler = resendSigninLinkHandler;
+
+      break;
+    default:
+      throw new Error('Invalid link type submitted to LinkExpired component');
+  }
+  return templateValues;
+};
+
+const LinkExpired = ({ linkType }: LinkExpiredProps) => {
+  // TODO : Metric event(s) for expired link
+
+  const templateValues = getTemplateValues(linkType);
+
+  return (
+    <>
+      {/* TODO: Add alertBar for success/failure status of resendLinkHandler */}
+      <FtlMsg id={templateValues.headerId}>
+        <h1 id="fxa-link-expired-header" className="card-header">
+          {templateValues.headerText}
+        </h1>
+      </FtlMsg>
+
+      <FtlMsg id={templateValues.messageId}>
+        <p className="mt-4 text-sm">{templateValues.messageText}</p>
+      </FtlMsg>
+      <FtlMsg id="resend-link">
+        <button
+          onClick={templateValues.resendLinkHandler}
```

Comment:
If we don't go the FtlResolver.Msg route, maybe we can at least pass in the handler separately? Putting it on `templateValues` just feels a little funky to me.

Reply:
sure!